### ISSUE TITLE
type mistaken in class inheritance

### ISSUE THREAD

**Issue by xiaochuan-li** [PyCQA/pylint#3805]
hello I ran into some bugs when I was trying to deal with class inheritance. I used the code as below. 
```python
class parent:
    def __init__(self):
        print("this is parent from init")

    @classmethod
    def _construct(cls):
        print("this is parent from cmeth")
        return cls()


class child(parent):
    def __init__(self):
        super(child, self).__init__()
        print("this is child from init")

    @classmethod
    def _construct(cls):
        print("this is child from cmeth")
        return super(child, cls)._construct()

    def child_meth(self):
        print("this is child meth")


if __name__ == "__main__":
    c = child()
    print(type(c))
    print("#" * 10)
    c1 = child._construct()
    c1.child_meth()
    print(type(c1))
```

### Current behavior
the type of c1 was detected as `parent` by pylint, therefore it showed me the alert: `Instance of 'parent' has no 'child_meth' memberpylint(no-member)`

### Expected behavior
however as the output showed that the type of c1 was `"__main__.child"`

### pylint --version output
I am using pylint as an extension of vs code 
Thank you
